client/src/components/UsersPost.tsx

```diff
@@ -24,31 +29,90 @@ interface Post {
   title: string;
   body: string;
 }
+
 interface PostProps {
   post: Post;
+  getPosts: () => void;
 }
 
-const UsersPost = ({ post }: PostProps): React.ReactElement => {
+const UsersPost = ({ post, getPosts }: PostProps): React.ReactElement => {
   const [like, setLike] = useState(false);
+  const [edit, setEdit] = useState(false);
+  const [editedPost, setEditedPost] = useState({ ...post });
 
   const handleLike = () => {
     setLike(!like);
   };
 
+  const handleEdit = () => {
+    setEdit(true);
+  };
+
+  const handleTitleChange = (e: any) => {
+    setEditedPost({ ...editedPost, title: e.target.value });
+  };
+
+  const handleBodyChange = (e: any) => {
+    setEditedPost({ ...editedPost, body: e.target.value });
+  };
```

Comment:
`handleBodyChange` and `handleTitleChange` can be combined into a single function see [PostCreationPage](./PostCreationPage.tsx) `handleTextInput`